From 2e774932be677d2c1a1b2695c2f06a6c85f31b7f Mon Sep 17 00:00:00 2001
From: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
Date: Tue, 11 Aug 2026 19:03:19 -0300
Subject: [PATCH] dt-bindings: watchdog: airoha: add EN751221

The EcoNet EN751221 watchdog uses the same timer/watchdog register
layout and BUS/2 input clock as the Airoha EN7581 block. Allow the
EN751221 compatible in the existing binding.

Signed-off-by: OpenAI <noreply@openai.com>
Signed-off-by: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
---
 .../devicetree/bindings/watchdog/airoha,en7581-wdt.yaml     | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Documentation/devicetree/bindings/watchdog/airoha,en7581-wdt.yaml b/Documentation/devicetree/bindings/watchdog/airoha,en7581-wdt.yaml
index 6bbab3cb28e5..65214d203916 100644
--- a/Documentation/devicetree/bindings/watchdog/airoha,en7581-wdt.yaml
+++ b/Documentation/devicetree/bindings/watchdog/airoha,en7581-wdt.yaml
@@ -4,7 +4,7 @@
 $id: http://devicetree.org/schemas/watchdog/airoha,en7581-wdt.yaml#
 $schema: http://devicetree.org/meta-schemas/core.yaml#
 
-title: Airoha EN7581 Watchdog Timer
+title: Airoha and EcoNet Watchdog Timer
 
 maintainers:
   - Christian Marangi <ansuelsmth@gmail.com>
@@ -14,7 +14,9 @@ allOf:
 
 properties:
   compatible:
-    const: airoha,en7581-wdt
+    enum:
+      - airoha,en7581-wdt
+      - econet,en751221-wdt
 
   reg:
     maxItems: 1
-- 
2.55.0

